src/oop/programs_with_multiple_classes/medium/q5/MainTest.java

```diff
@@ -0,0 +1,122 @@
+                {343, true},
+                {3434, false},
+                {7653, true}
+        };
+        Class<?>[] classArguments = {int.class, boolean.class};
+        assertTrue(monitor.hasConstructor(classArguments), modifiedMonitorMessage);
+        assertTrue(monitor.hasModifier(classArguments, "public"), modifiedMonitorMessage);
+        for (int i = 0; i < tests.length; i++) {
+            Object[][] arguments = {
+                    {tests[i][0], int.class},
+                    {tests[i][1], boolean.class}
+            };
+            Object chairInstance = monitor.createInstance(arguments);
+            _assertEquals(tests[i][0], monitor.getFieldValue(chairInstance, varCount),
+                    "Your " + monitorClass + " constructor does not correctly initialize the " + varCount + " field.");
+            _assertEquals(tests[i][1], monitor.getFieldValue(chairInstance, varOn),
+                    "Your " + monitorClass + " constructor does not correctly initialize the " + varOn + " field.");
+        }
+    }
+
+    @Test
+    public void deskClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {String.class};
+        assertTrue(desk.hasConstructor(classArguments),
+                "Your " + deskClass + " constructor does not have the correct parameters.");
+        assertTrue(desk.hasModifier(classArguments, "public"),
+                "Your " + deskClass + " constructor does not have the correct modifier.");
+    }
+
+    private static Stream<Arguments> deskInputProvider() {
+        return Stream.of(
+                Arguments.of("Mialani"),
+                Arguments.of("Oak"),
+                Arguments.of("Steel")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("deskInputProvider")
+    public void deskConstructorInitializesValuesCorrectly(String material) throws Throwable {
+        Object[][] arguments = {
+                {material, String.class},
+        };
+        Object chairInstance = desk.createInstance(arguments);
+        _assertEquals(material, desk.getFieldValue(chairInstance, varMaterial),
+                "Your " + deskClass + " constructor does not correctly initialize the " + varMaterial + " field.");
+        assertEquals(null, desk.getFieldValue(chairInstance, varMonitors),
+                "Your " + deskClass + " constructor does not correctly initialize the " + varMonitors + " field.");
+    }
+
+    private static Stream<Arguments> monitorInputProvider() {
+        return Stream.of(
+                Arguments.of(11, true),
+                Arguments.of(20, false),
+                Arguments.of(43, true)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("monitorInputProvider")
+    public void correctFallenMethod(int count, boolean on) throws Throwable {
+        Object[][] arguments = {
+                {count, int.class},
+                {on, boolean.class}
+        };
+        Object monitorInstance = monitor.createInstance(arguments);
```

Comment:
Please also test the return type of the method separately. You can do this with `hasMethod`, and specify the return type to be `Void.TYPE`